kernel/kernel-api/src/main/java/io/delta/kernel/internal/metrics/TransactionMetrics.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.metrics;
+
+import io.delta.kernel.metrics.TransactionMetricsResult;
+
+/**
+ * Stores the metrics for an ongoing transaction. These metrics are updated and recorded throughout
+ * the transaction using this class.
+ *
+ * <p>At report time, we create an immutable {@link TransactionMetricsResult} from an instance of
+ * {@link TransactionMetrics} to capture the metrics collected during the transaction. The {@link
+ * TransactionMetricsResult} interface exposes getters for any metrics collected in this class.
+ */
+public class TransactionMetrics {
+
+  public final Timer totalCommitTimer = new Timer();
+
+  public final Counter commitAttemptsCounter = new Counter();
+
+  public final Counter addFilesCounter = new Counter();
+
+  public final Counter removeFilesCounter = new Counter();
+
+  public final Counter totalActionsCounter = new Counter();
+
+  /**
+   * Resets the action counters (addFilesCounter, removeFilesCounter and totalActionsCounter) to 0
```

Comment:
Could you add a bit of context _why_ someone would want to reset them?

Reply:
Added